src/district0x/ui/events.cljs

```diff
@@ -173,6 +173,30 @@
     (assoc-in db [:config] config)))
 
 (reg-event-fx
+  :district0x.browsing/setup!
+  interceptors
+  (fn [{:keys [db]} [routes]]
+    (if (contains? (-> db
+                       (get-in [:config :pushroute-hosts])
+                       set)
+                   (d0x-ui-utils/current-host))
+      {:db (assoc-in db [:browsing :hashroutes?] false)
+       :district0x.browsing/pushroutes! routes}
+      {:db (assoc-in db [:browsing :hashroutes?] true)
+       :district0x.browsing/hashroutes! routes})))
+
+(reg-fx
+  :district0x.browsing/pushroutes!
```

Comment:
Convention is to not put ! `reg-fx` or `reg-event-fx`, because they're known to do side effects. Anyways I think better names would be  `:district0x.browsing/setup-pushroutes`, `:district0x.browsing/setup-hashroutes`. And one above can be `:district0x.browsing/setup`